### PR TITLE
feat(stepfunctions-tasks): add validations for EmrCreateCluster

### PR DESCRIPTION
**Pull request by go-to-k**
This PR adds validations for `EmrCreateCluster`.

## timeoutDurationMinutes

> Minimum value is 5 and maximum value is 1440. 

https://docs.aws.amazon.com/emr/latest/APIReference/API_SpotProvisioningSpecification.html

## bidPrice and bidPriceAsPercentageOfOnDemandPrice

Both `bidPrice` and `bidPriceAsPercentageOfOnDemandPrice` are specified, the error occurs in Step Functions console.

```
Specify at most one of bidPrice or bidPriceAsPercentageOfOnDemandPrice value for the Spot Instance fleet : Master request. (Service: AmazonElasticMapReduce; Status Code: 400; Error Code: ValidationException; Request ID: xxxxxx; Proxy: null)
```

## targetOnDemandCapacity and targetSpotCapacity

> At least one of TargetSpotCapacity and TargetOnDemandCapacity should be greater than 0. For a master instance fleet, only one of TargetSpotCapacity and TargetOnDemandCapacity can be specified, and its value must be 1.

https://docs.aws.amazon.com/emr/latest/APIReference/API_InstanceFleetConfig.html

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk-testing/framework-integ/test/aws-lambda-nodejs/test/integ.latest.js.snapshot/LambdaNodeJsLatestIntegDefaultTestDeployAssertD40B5C28.assets.json
Comment (go-to-k) [inline packages/@aws-cdk-testing/framework-integ/test/aws-lambda-nodejs/test/integ.latest.js.snapshot/LambdaNodeJsLatestIntegDefaultTestDeployAssertD40B5C28.assets.json:1]: Although not related to this PR, CodeBuild was causing errors due to differences caused by other PRs, so I changed it.

Comment (go-to-k) [reply packages/@aws-cdk-testing/framework-integ/test/aws-lambda-nodejs/test/integ.latest.js.snapshot/LambdaNodeJsLatestIntegDefaultTestDeployAssertD40B5C28.assets.json:1]: I have submitted another PR. If it gets merged, I will pull it in here as well.

https://github.com/aws/aws-cdk/pull/28530

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/private/cluster-utils.ts [hunk 1]
```diff
@@ -146,6 +150,9 @@ function SpotProvisioningSpecificationPropertyToJson(property?: EmrCreateCluster
   if (!property) {
     return undefined;
   }
+  if (!cdk.Token.isUnresolved(property.timeoutDurationMinutes) && (property.timeoutDurationMinutes < 5 || property.timeoutDurationMinutes > 1440)) {
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/private/cluster-utils.ts:153]: Unrelated to your PR, but I can't believe we have a prop called `timeoutDurationMinutes: number`. It should have been typed and named as `timeout: Duration`. Wonder if you're willing to deprecate `timeoutDurationMinutes` in favor of a new prop in a separate PR

**Review by kaizencc — APPROVED** [review]
Thanks :)

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/private/cluster-utils.ts (hunk 1 above)
Comment (go-to-k) [reply packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/private/cluster-utils.ts:153]: @kaizencc 

> Wonder if you're willing to deprecate timeoutDurationMinutes in favor of a new prop in a separate PR

OK, I will take it.

Comment (go-to-k) [reply packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/private/cluster-utils.ts:153]: @kaizencc 

Oh, this param `timeoutDurationMinutes` is required. This means that even if we have a new parameter, we still need to specify this parameter, and making it optional would be a breaking change.

What should we do?

Comment (go-to-k) [reply packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/private/cluster-utils.ts:153]: @kaizencc 

Oh, this param `timeoutDurationMinutes` is required in the interface. ~~This means that even if we have a new parameter, we still need to specify this parameter, and making it optional would be a breaking change.~~

I was mistaken: making a required parameter optional for an interface that is a props is not a breaking change.

But making the new parameter require is a breaking change.

What should we do? Should it be added as a new optional parameter, whereas before it was required?

Comment (go-to-k) [reply packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/private/cluster-utils.ts:153]: I'll submit a PR anyway.

Comment (go-to-k) [reply packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/private/cluster-utils.ts:153]: @kaizencc 

I have submitted the PR.

https://github.com/aws/aws-cdk/pull/28532#issue-2060755266

**Pull request merged**